test/test_metapackage.py

```diff
@@ -76,6 +86,16 @@ def test_validate_metapackage(self):
                 else:
                     _validate_metapackage(path, package)
 
+    def test_collect_warnings(self):
+        """Tests warnings collection"""
+        warnings = []
+        pkgs_dict = find_packages(test_data_dir, warnings=warnings)
+
+        warnings.sort()
+        test_expected_warnings.sort()
+        for w, we in zip(warnings, test_expected_warnings):
+            assert w == we, 'Warning message "%s" != "%s"' % (w,we)
```

Comment:
But by that logic he should do sorted lists, or additionally assert that the set is equal to the original list. Comparing sets alone will not catch one list having duplicates that the other does not.

I still think my previous comment applies:

https://github.com/ros-infrastructure/catkin_pkg/pull/114#discussion_r26631509

I'd recommend using `self.assertEqual` on either the lists or the sorted version of the lists, because if one warning doesn't match, it will most likely require the context of the list of warnings to figure out what is the issue.